DOC: Kernel density notebook update

tommyod opened the pull request:
First PR to statsmodels. The Jupyter Notebook on Kernel Density Estimation seemed very incomplete to me, so I updated it with the following:

- More text and comment explaining the code.
- Example comparing a true distribution to the KDE of the sampled distribution.
- The effect of varying the `bw` argument (the bandwidth of the kernel).
- The various kernel functions available (one plot for each of them).
- Cleaned up the plots a bit: smaller, with grids, default maplotlib colors instead of black/red.

I also fixed a minor typo.

Comments very welcome!

josef-pkt commented:
@tommyod  Thanks for the improvement

I only had time for a very quick skimming of it.
Looks good overall
content of notebook will need to be removed before merging

the only comment directly to the content is that icdf doesn't fit in the plot with cdf and sf.
The x-axis for icdf is misleading because it should be the interval [0, 1]

tommyod commented:
@josef-pkt Thanks for the review. I read through again and wrote a little bit more. I removed the output of the Jupyter Notebook, and addressed the comment about the domain of the inverse CDF as per your request :)

josef-pkt commented on examples/notebooks/kernel_density.ipynb:
```diff
@@ -28,208 +41,423 @@
+    "dist1_loc, dist1_scale, weight1 = -1 , .5, .25\n",
+    "dist2_loc, dist2_scale, weight2 = 1 , .5, .75\n",
+    "\n",
+    "# Sample from a micture of distributions\n",
```
typo `mixture`

josef-pkt commented:
merged in #4551 

@tommyod  Thanks for the improvement
